txdbus/authentication.py

```diff
@@ -315,11 +320,13 @@ def _step_one(self, username, keyring_dir=None):
 
         self.challenge_str = binascii.hexlify(hashlib.sha1(
                                               os.urandom(8)).digest())
-        
-        msg = ' '.join( [self.cookieContext,
-                         str(self.cookieId),
-                         self.challenge_str] )
-        
+
+        msg = '{} {} {}'.format(
+            self.cookieContext,
+            self.cookieId,
+            self.challenge_str.decode(),
+        )
```

Comment:
This PR still needs a bit more refinement. My first goal is to get everything 100% working, and then clean up these sort of double-casts, etc (it's easier to refactor and clean up once all tests pass).